update: smacss order

Yimiprod opened the pull request:
https://github.com/brigade/scss-lint/issues/883

trotzig commented:
This is quite a big change to the order. Could you explain the reasoning a little? For instance, you've moved z-index from the "other" group into the "box" group.

Yimiprod commented:
I'm not sure it's the good way, but since there's no reference about ordering this rules, in particular the **others**, it's based on way people tend to write thoses rules.

Some of those choices certainly deserve more discution and more people to talk about those orders.

**Why moving _other_ properties :**
- **content** on first position, almost only used in pseudo-elements, that way it's the first properties for those elements and never forgot to put that propertie.
- **quotes** just after, same reasoning as **content**, since it's used in specific element.
- **visibility**, since it's a display propertie, certainly deserve more reflection
- **z-index**, some people tend to like it at last position, it's arguable, but it's a position propertie
- **overflow**, missing x and y, it was let in other, but it certainly belong to box model
- **table-layout** was missing, it's a box model
- **animations** probably at the wrong place in this order, i'm not sure about that point, but they were missing
- **stroke** was missing in border properties, it's the border of an svg element
- **opacity** was left in other, moved it in background since it doesn't change box model but doesn't change the display, it subjectif
- **box-shadow** should belong with background, it change the way the box appear
- **fill** was missing, it's the background of an svg element
- **text** many font properties was missing, ordered mainly by name.

I hope it's clear, i've added some new properties in other, not knowing where to add thems, maybe deserve a User interaction section.

I'm open to any changes, and if there's a place to talk about properties ordering, it would be good to have the link.

trotzig commented:
Thanks. We're not using this rule ourselves at Brigade, so it would be good to have some outside feedback by people who use it. 

I'm pinging a few people who have made contributions to smacss in the past, to see if they have any input: @danny-andrews, @tobiashm, @bkzl, @ingdir, @tgecho.

bkzl commented:
I like the changes proposed by @Yimiprod and agree with all of the above. +1

tobiashm commented on data/property-sort-orders/smacss.txt:
```diff
@@ -1,15 +1,23 @@
 # SMACSS Property Order
 # http://smacss.com/book/formatting
 
+content
```
I think it would be a good idea to precede this with a comment about these properties being outside of the groups listed at https://smacss.com/book/formatting#grouping (sine the following rules are grouped by a comment “heading”)

Yimiprod replied:
Will do. Thanks

tobiashm commented on data/property-sort-orders/smacss.txt:
```diff
@@ -121,41 +167,66 @@ color
+font-smooth
 font-smoothing
+-moz-osx-font-smoothing
+-webkit-font-smoothing
```
There're no other vendor prefixed rules in these lists, so I don't think it's a good idea to start adding these here.

Yimiprod replied:
Well to be fair, i added them because those two properties are mandatory when doing a custom font and it's commonly used.
But i think i'll remove them, since it's not standard at the moment.

trotzig replied:
Good call. I suggest you look into using something like [autoprefixer](https://github.com/postcss/autoprefixer) to avoid having to use vendor-prefixes.

Yimiprod replied:
I'm thinking about deleting **font-smooth** and **font-smoothing** since they aren't standard.

tobiashm commented:
I think this looks pretty good. There're not many specifics listed at https://smacss.com/book/formatting#grouping except for the general grouping principles. So maybe it would be a good idea to add some comments about this being _based on_ SMACSS, with some additional principles for property orders, just so that others don't take this to be an official expression of Jonathan Snooks intent.

Yimiprod commented:
Maybe i should ask him what he think about it?

Yimiprod commented:
@snookca look not active on github, since i don't want to lure people, i'll ask him in a mail.
If no i get no anwser i'll add the comments.

Thank for the reviews.

CeliaLfd commented:
Following the proposal of @Yimiprod, it would be a good idea to think about the properties `grid-layout` that will be supported soon.

Yimiprod commented:
Got a response by mail from @snookca where he seem to aggree
> Happy to hear the book has been helpful!
> I took a look at that PR. No specific objections with what you’ve outlined. All makes sense to me. :) 
![image](https://cloud.githubusercontent.com/assets/2227867/22649133/f69f79a6-ec79-11e6-98cc-2fa731f56908.png)

trotzig commented:
The Travis failure is for the jruby build, most likely not related to your changes. Thanks @Yimiprod for contributing to this project!
